### PR TITLE
[WIP] RBAC architecture description

### PR DESCRIPTION
**Pull request by Fryguy**
Rendered: https://github.com/Fryguy/guides/blob/376c6b0a55e46be9e931d4152366cea9190f6f0d/architecture/rbac.md

### CODE REVIEW

File: architecture/rbac.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# RBAC
+
+![RBAC](images/rbac.svg)
+
+## How it works
+
+RBAC in ManageIQ is split into 2 parts; what you can "see" and what you can "do".  All user interaction first pass through the first layer of things you can "see" based on a number of criteria.  On those objects, then, only a certain subset of actions are allowed based on what you can "do".
```
Comment (jrafanie) [inline architecture/rbac.md]: >All user interaction first pass through the first layer of things you can "see" based on a number of criteria

I'm not sure what you're trying to say here with wit the use "first" and "layers".  99% of user interaction involves users being shown objects they are allowed to see.  We then enable/disable various actions in menus based on what you can do.

Comment (Fryguy) [reply architecture/rbac.md]: Yeah good point.  Later, the "layers" are enumerated, but here it's premature.  I'll reword.

File: architecture/rbac.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+At the heart of the RBAC system are Users and Groups.  A user can belong to multiple groups, and while a user is interacting with ManageIQ there is the notion of a "current" group.  Only the current group's permission set will apply during that session, or until the "current" group is changed.  Both a user and a group can directly "own" resources in the system.
```
Comment (jrafanie) [inline architecture/rbac.md]: > Both a user and a group can directly "own" resources in the system.

... which means the current group can also change what you can "see".

File: architecture/rbac.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+- Ownership
+
+  A resource in ManageIQ can be directly owned by a User or Group.  A resource can additionally be owned by a tenant.  All objects directly owned by a user or owned by the user's current group can be seen.
+
+- Tenancy
+
+  Tenancy ownership in ManageIQ is determined by first a direct ownership to a resource, and then applying certain "ancestor" or "descendant" rules depending on the type of resource being viewed.
+
+  For example, a member of a tenant can only see virtual templates owned by their tenant or owned by any parent tenants.  This "ancestor" view in used to allow for sharing a virtual template across multiple child tenants.  In contrary, a member of a tenant can only see virtual machine instances owned by their tenant or owned by any child tenants.  This "descendant" view is used to allow the parent tenant to do accounting of all running virtual machines instances below them.
```
Comment (jrafanie) [inline architecture/rbac.md]: In contrast

File: architecture/rbac.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    A user can be given access to various resources by allowing them to view tagged resources.  A user is given access to a "managed tag", and any resource tagged with it can be seen.
+
+  - belongs to
+
+    A user can be given access to various resources based on the resource hierarchy from the provider.  For example, a user can be given access to a virtual infrastructure Cluster.  In doing, so any child host or virtual machine instance that "belongs to" that Cluster can be see by the user.
+
+  These two filters combine together in such a way that if both are specifed they constrain each other, otherwise all resources can be seen.  That is,
+
+  - if a managed tag is specified without a belongs to, then all tagged resources can be seen
+  - if a belongs to is specified without a managed tag, then all resources in the resource hierarchy can be seen.
```
Comment (jrafanie) [inline architecture/rbac.md]: @bdunne's 2nd favorite belong-sto

**Comment by Fryguy** [comment]
Closing in favor of #412

**Pull request closed without merging**